rust/src/data_catalog/unity/mod.rs

```diff
@@ -75,17 +36,416 @@ pub enum GetTableError {
+        // https://docs.databricks.com/api-explorer/workspace/schemas/get
+        let resp = self
+            .client
+            .get(format!(
+                "{}/schemas/{}.{}",
+                self.catalog_url(),
+                catalog_name.as_ref(),
+                schema_name.as_ref()
+            ))
+            .header(AUTHORIZATION, token)
+            .send_retry(&self.retry_config)
+            .await?;
+        Ok(resp.json().await?)
+    }
+
+    /// Gets an array of summaries for tables for a schema and catalog within the metastore.
+    ///
+    /// The table summaries returned are either:
+    /// - summaries for all tables (within the current metastore and parent catalog and schema),
+    ///   when the user is a metastore admin, or:
+    /// - summaries for all tables and schemas (within the current metastore and parent catalog)
+    ///   for which the user has ownership or the SELECT privilege on the table and ownership or
+    ///   USE_SCHEMA privilege on the schema, provided that the user also has ownership or the
+    ///   USE_CATALOG privilege on the parent catalog.
+    ///
+    /// There is no guarantee of a specific ordering of the elements in the array.
+    pub async fn list_table_summaries(
+        &self,
+        catalog_name: impl AsRef<str>,
+        schema_name_pattern: impl AsRef<str>,
+    ) -> CatalogResult<ListTableSummariesResponse> {
+        let token = self.get_credential().await?;
+        // https://docs.databricks.com/api-explorer/workspace/tables/listsummaries
+        let resp = self
+            .client
+            .get(format!("{}/table-summaries", self.catalog_url()))
+            .query(&[
+                ("catalog_name", catalog_name.as_ref()),
+                ("schema_name_pattern", schema_name_pattern.as_ref()),
+            ])
+            .header(AUTHORIZATION, token)
+            .send_retry(&self.retry_config)
+            .await?;
+
+        Ok(resp.json().await?)
+    }
+
+    /// Gets a table from the metastore for a specific catalog and schema.
+    ///
+    /// The caller must be a metastore admin, be the owner of the table and have the
+    /// USE_CATALOG privilege on the parent catalog and the USE_SCHEMA privilege on
+    /// the parent schema, or be the owner of the table and have the SELECT privilege on it as well.
+    ///
+    /// # Parameters
+    pub async fn get_table(
+        &self,
+        catalog_id: Option<impl AsRef<str>>,
+        database_name: impl AsRef<str>,
+        table_name: impl AsRef<str>,
+    ) -> CatalogResult<GetTableResponse> {
+        let token = self.get_credential().await?;
+        // https://docs.databricks.com/api-explorer/workspace/tables/get
+        let resp = self
+            .client
+            .get(format!(
+                "{}/tables/{}.{}.{}",
+                self.catalog_url(),
+                catalog_id
+                    .map(|c| c.as_ref().to_string())
+                    .unwrap_or("main".into()),
```

Comment:
You should be able to do this instead:
```suggestion
                    .as_ref()
                    .map(|c| c.as_ref())
                    .unwrap_or("main"),
```